monai/losses/dice.py

```diff
@@ -84,41 +94,75 @@ def forward(self, input: torch.Tensor, target: torch.Tensor, smooth: float = 1e-
                 warnings.warn("single channel prediction, `include_background=False` ignored.")
         else:
             if self.do_softmax:
-                input = torch.softmax(input, 1)
+                input = torch.softmax(input, dim=1)
             if self.to_onehot_y:
-                target = one_hot(target, n_pred_ch)
+                target = one_hot(target, num_classes=n_pred_ch)
             if not self.include_background:
                 # if skipping background, removing first channel
                 target = target[:, 1:]
                 input = input[:, 1:]
+
         assert (
             target.shape == input.shape
         ), f"ground truth has differing shape ({target.shape}) from input ({input.shape})"
 
         # reducing only spatial dimensions (not batch nor channels)
         reduce_axis = list(range(2, len(input.shape)))
-        intersection = torch.sum(target * input, reduce_axis)
+        intersection = torch.sum(target * input, dim=reduce_axis)
 
         if self.squared_pred:
             target = torch.pow(target, 2)
             input = torch.pow(input, 2)
 
-        ground_o = torch.sum(target, reduce_axis)
-        pred_o = torch.sum(input, reduce_axis)
+        ground_o = torch.sum(target, dim=reduce_axis)
+        pred_o = torch.sum(input, dim=reduce_axis)
 
         denominator = ground_o + pred_o
 
         if self.jaccard:
             denominator -= intersection
 
         f = 1.0 - (2.0 * intersection + smooth) / (denominator + smooth)
-        if self.reduction == "sum":
-            return f.sum()  # sum over the batch and channel dims
-        if self.reduction == "none":
-            return f  # returns [N, n_classes] losses
+
         if self.reduction == "mean":
-            return f.mean()  # the batch and channel average
-        raise ValueError(f"reduction={self.reduction} is invalid.")
+            f = torch.mean(f)  # the batch and channel average
+        elif self.reduction == "sum":
+            f = torch.sum(f)  # sum over the batch and channel dims
+        elif self.reduction == "none":
+            pass  # returns [N, n_classes] losses
+        else:
+            raise ValueError(f"reduction={self.reduction} is invalid.")
+
+        return f
+
+
+class MaskedDiceLoss(DiceLoss):
+    """
+    Same as DiceLoss, but accepts a binary mask ([0,1]) indicating a region over which to compute the dice.
+    """
+
+    def forward(self, input: torch.Tensor, target: torch.Tensor, mask: torch.Tensor = None, smooth: float = 1e-5):
+        """
+        Args:
+            input (tensor): the shape should be BNH[WD].
+            target (tensor): the shape should be BNH[WD].
+            mask (tensor): (optional) the shape should B1H[WD]
```

Comment:
I think the `mask` supports the broadcasting rules by construction, such as  `BNH[WD]`, `BN1[, 1, 1]`, `BNH[W, 1]`... the docstring needs to be updated

Reply:
we'll support only B1H[WD] and BNH[WD]. added text